File: chapter 4 - quicksort/quicksort.py
def quicksort(list):
    if len(list) < 2:
        return list
    else:
        pivot = list[0]
        less = [i for i in list[1:] if i < pivot]
        greater = [i for i in list[1:] if i > pivot]

        return quicksort(less) + [pivot] + quicksort(greater)


print(quicksort([3, 4, 62, 1, 7, 4, 9, 0]))

File: chapter 4 - quicksort/quicksort.js
function quicksort(arr) {
  if (arr.length < 2) {
    return arr
  } else {
    const pivot = arr[0]
    const less = arr.slice(1).filter(item => item < pivot)
    const greater = arr.slice(1).filter(item => item > pivot)

    return quicksort(less).concat([pivot], quicksort(greater))
  }
}

console.log(quicksort([3, 4, 62, 1, 7, 4, 9, 0]))
